node/pkg/logscribe/app.go

```diff
@@ -87,3 +131,106 @@ func bulkCopyLogs(ctx context.Context, logsChannel <-chan *[]api.LogInsertModel)
 		}
 	}
 }
+
+func (a *App) processLogs(ctx context.Context) {
+	ticker := time.NewTicker(a.processLogsInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+			return
+		case <-ticker.C:
+			processedLogs := make([]LogInsertModelWithID, 0)
+			logMap := make(map[string]bool, 0)
+			totalLogs := 0
+			for {
+				logs, err := fetchLogs(ctx)
+				if err != nil || len(logs) == 0 {
+					break
+				}
+				totalLogs += len(logs)
+
+				for _, log := range logs {
+					hash := hashLog(log)
+					if !logMap[hash] {
+						processedLogs = append(processedLogs, log)
+						logMap[hash] = true
+					}
+				}
+
+				err = deleteLogs(ctx)
+				if err != nil {
+					break
+				}
```

Comment:
if it is defined to be deleted after reading, it'd be better to update pgs query
```
`DELETE FROM logs WHERE id IN (SELECT id FROM logs LIMIT @limit) RETURNING *;`
```

you won't need to separate delete and read, but by just calling this delete query you'll be able to read all deleted logs